test/jdk/com/sun/net/httpserver/BasicAuthToken.java

```diff
@@ -0,0 +1,117 @@
+ * @run main/othervm BasicAuthToken
+ */
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.util.Base64;
+import com.sun.net.httpserver.Authenticator;
+import com.sun.net.httpserver.BasicAuthenticator;
+import com.sun.net.httpserver.HttpServer;
+
+public class BasicAuthToken {
+    private static final String CRLF = "\r\n";
+    private static final String someContext = "/test";
+
+    public static void main(String[] args) throws Exception {
+        HttpServer server = server();
+        try {
+            client(server.getAddress().getPort());
+        } finally {
+            server.stop(0);
+        }
+    }
+
+    static HttpServer server() throws Exception {
+        String realm = "someRealm";
+        ServerAuthenticator authenticator = new ServerAuthenticator(realm);
+        HttpServer server = HttpServer.create(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0), 0);
+        server.createContext(someContext, exchange -> {
+            if (authenticator.authenticate(exchange) instanceof Authenticator.Failure) {
+                exchange.sendResponseHeaders(401, -1);
+                exchange.close();
+                return;
+            }
+            exchange.sendResponseHeaders(200, -1);
+            exchange.close();
+        }).setAuthenticator(authenticator);
+        server.start();
+        return server;
+    }
+
+    static void client(int port) throws Exception {
+        try (Socket socket = new Socket(InetAddress.getLoopbackAddress(), port)) {
+            BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(socket.getOutputStream()));
+            BufferedReader reader = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+            String credentials = "username:password";
+            String encodedCredentials = Base64.getEncoder().encodeToString(credentials.getBytes());
+            writer.write("GET " + someContext + " HTTP/1.1" + CRLF);
+            writer.write("Host: localhost:" + port + CRLF);
+            writer.write("User-Agent: Java/" + System.getProperty("java.version") + CRLF);
+            writer.write("Authorization: BAsIc " + encodedCredentials + CRLF);
+            writer.write(CRLF);
+            writer.flush();
+
+            System.err.println("Server response");
+            String statusLine = reader.readLine();
+            System.err.println(statusLine);
+
+            if (!statusLine.startsWith("HTTP/1.1 200")) {
+                throw new RuntimeException("Basic Authentication failed: case-insensitive token" +
+                        " used to identify authentication scheme  sent by client parsed incorrectly");
+            }
+            assert ServerAuthenticator.wasChecked() : "Authenticator was not correctly invoked";
```

Comment:
As far as I know, the jtreg tests that we launch (through make) will always have asserts enabled. So I think using `assert` here is OK. However, to be consistent with other conditional checks in this test, I think it would be better to change this to a if block, something like:

```java
if (!ServerAuthenticator.wasChecked()) { 
   throw new RuntimeException("Authenticator wasn't invoked");
}
```

Reply:
Fixed it! And yes, all tests passed using `assert`